templates/content-certificate.php

```diff
@@ -13,7 +13,11 @@
 $cert = new LLMS_User_Certificate( get_the_ID() );
 $uid  = get_current_user_id();
 
-if ( $uid != $cert->get_user_id() && ! llms_can_user_bypass_restrictions( $uid ) ) {
+$is_owner_or_admin 		= ($uid == $cert->get_user_id() || llms_can_user_bypass_restrictions( $uid ));
+$is_shareable					= (get_post_meta( get_the_ID(), '_llms_allow_sharing', true ) === 'yes');
+$is_allow_to_display	= $is_owner_or_admin || $is_shareable;
```

Comment:
I think this logic can be made more generally useful (possibly internally in the future but certainly for any custom development) by moving it into class methods on the `LLMS_User_Certificate` model.

I envision two methods (pseduo code below):

```php
public function can_user_manage( $user_id = null ) {
 
  $user_id = $user_id or current user

  return existing logic of  $is_owner_or_admin var in PR

}
```

and

```php
public function can_user_view( $user_id = null ) {
 
  $user_id = $user_id or current user
  
  return $this->can_user_manage() || llms_parse_bool( $this->get( 'allow_sharing' ) )

}
```

Both methods should additionally have filters on their returns!